src/include/distributed/shared_connection_stats.h

```diff
@@ -14,15 +14,18 @@
 #define ADJUST_POOLSIZE_AUTOMATICALLY 0
 #define DISABLE_CONNECTION_THROTTLING -1
 #define DISABLE_REMOTE_CONNECTIONS_FOR_LOCAL_QUERIES -1
+#define ALLOW_ALL_EXTERNAL_CONNECTIONS -1
```

Comment:
Probably rename this too, to better fit the new GUC name:
```suggestion
#define ALLOW_ALL_CLIENT_CONNECTIONS -1
```

 (feel free to come up with an better name than the one I propose here)

Reply:
the CLIENT part has the same meaning, right? That's why we call the GUC max_client_connections. Maybe call it UNLIMITED_CLIENT_CONNECTIONS.

Reply:
maybe as a user facing API, `citus.max_client_connections` is good because users are not (and probably should not) know about the internal connections of Citus. They only care about their own connections to the database, which is  a `client`

But at this point, internal Citus backends are also `clients` from Postgres' perspective and calling it only `client` slightly confused me.

(Still, the proposed change would not hurt, but I like `external` a little more)

Reply:
> UNLIMITED_CLIENT_CONNECTIONS

Maybe `ALLOW)UNLIMITED_EXTERNAL_CONNECTIONS`

Reply:
unlimited kind of sounds wrong, since it's not unlimited